opennms-provision/opennms-detector-simple/src/main/java/org/opennms/netmgt/provision/detector/snmp/GenericSnmpDetectorFactory.java

```diff
@@ -62,7 +67,22 @@ public DetectRequest buildRequest(String location, InetAddress address, Integer
         if (m_agentConfigFactory == null) {
             throw new IllegalStateException("Cannot determine agent configuration without a SnmpAgentConfigFactory.");
         }
-        return m_agentConfigFactory.getAgentConfig(address, location).toMap();
+        List<SnmpProfile> snmpProfiles = m_agentConfigFactory.getProfiles();
+        if (snmpProfiles.isEmpty()) {
+            return m_agentConfigFactory.getAgentConfig(address, location).toMap();
+        } else {
+            Map<String, String> agentConfigMap = new HashMap<>();
+            agentConfigMap.put(AgentBasedSyncAbstractDetector.HAS_MULTIPLE_AGENT_CONFIGS, Boolean.toString(true));
```

Comment:
hasMultipleConfigs can be moved to DetectRequest, don't want to change this in point release.